aea/aea_builder.py

```diff
@@ -910,49 +936,142 @@ def get_connection_configuration(connection_id):
+                    )
             self._add_component_to_resources(contract)
 
     def _load_and_add_skills(self, context: AgentContext) -> None:
         for configuration in self._package_dependency_manager.skills.values():
             logger_name = "aea.packages.{}.skills.{}".format(
                 configuration.author, configuration.name
             )
-            skill_context = SkillContext()
-            skill_context.set_agent_context(context)
-            skill_context.logger = logging.getLogger(logger_name)
             configuration = cast(SkillConfig, configuration)
-            try:
-                skill = Skill.from_config(configuration, skill_context=skill_context)
-            except ModuleNotFoundError as e:
-                _handle_error_while_loading_component_module_not_found(configuration, e)
-            except Exception as e:
-                _handle_error_while_loading_component_generic_error(configuration, e)
+            if configuration in self._component_instances[ComponentType.SKILL].keys():
+                skill = cast(
+                    Skill, self._component_instances[ComponentType.SKILL][configuration]
+                )
+                skill.skill_context.set_agent_context(context)
+                skill.skill_context.logger = logging.getLogger(logger_name)
+            else:
+                skill_context = SkillContext()
+                skill_context.set_agent_context(context)
+                skill_context.logger = logging.getLogger(logger_name)
+                try:
+                    skill = Skill.from_config(
+                        configuration, skill_context=skill_context
+                    )
+                except ModuleNotFoundError as e:
+                    _handle_error_while_loading_component_module_not_found(
+                        configuration, e
+                    )
+                except Exception as e:
+                    _handle_error_while_loading_component_generic_error(
+                        configuration, e
+                    )
             self._add_component_to_resources(skill)
 
+    def _load_connection(
+        self, address: Address, configuration: ConnectionConfig
+    ) -> Connection:
+        """
+        Load a connection from a directory.
+
+        :param address: the connection address.
+        :param configuration: the connection configuration.
+        :return: the connection.
+        """
+        if configuration in self._component_instances[ComponentType.CONNECTION].keys():
+            connection = cast(
+                Connection,
+                self._component_instances[ComponentType.CONNECTION][configuration],
+            )
+            if connection.address != address:
+                logger.warning(
+                    "The address set on connection '{}' does not match the default address!".format(
+                        str(connection.connection_id)
+                    )
+                )
```

Comment:
we give a warning if the addresses don't match (the developer has to take care to use the correct address to instantiate the connection)...